File: maxhanna.Server/Controllers/DataContracts/Nexus/NexusBase.cs
using maxhanna.Server.Controllers.DataContracts.Users;

namespace maxhanna.Server.Controllers.DataContracts.Nexus
{
    public class NexusBase
    {
        public User? User { get; set; }
        public Decimal Gold { get; set; }
        public string? BaseName { get; set; }
        public int Supply { get; set; }
        public int CoordsX { get; set; }
        public int CoordsY { get; set; }
        public int CommandCenterLevel { get; set; }
        public int MinesLevel { get; set; }
        public int SupplyDepotLevel { get; set; }
        public int WarehouseLevel { get; set; }
        public int EngineeringBayLevel { get; set; }
        public int FactoryLevel { get; set; }
        public int StarportLevel { get; set; }
        public int MarineLevel { get; set; }
        public int GoliathLevel { get; set; }
        public int SiegeTankLevel { get; set; }
        public int ScoutLevel { get; set; }
        public int WraithLevel { get; set; }
        public int BattlecruiserLevel { get; set; }
        public int GlitcherLevel { get; set; }
        public DateTime Conquered { get; set; }
        public DateTime Updated { get; set; }
    }
}

File: maxhanna.client/src/services/datacontracts/nexus/nexus-base.ts
import { User } from "../user/user";

export interface NexusBase {
  user?: User;
  gold: number;
  supply: number;
  coordsX: number;
  coordsY: number;
  commandCenterLevel: number;
  minesLevel: number;
  supplyDepotLevel: number;
  engineeringBayLevel: number;
  warehouseLevel: number;
  factoryLevel: number;
  starportLevel: number;
  marineLevel: number;
  goliathLevel: number;
  siegeTankLevel: number;
  scoutLevel: number;
  wraithLevel: number;
  battlecruiserLevel: number;
  glitcherLevel: number;
}
